src/controllers/colormapperjsproxy.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <QScriptContext>
+#include <QScriptEngine>
+
+#include "controllers/colormapper.h"
+
+class ColorMapperJSProxy final : public QObject {
+    Q_OBJECT
+  public:
+    ColorMapperJSProxy() = delete;
+    ColorMapperJSProxy(QScriptEngine* pScriptEngine, QMap<QRgb, QVariant> availableColors);
+
+    ~ColorMapperJSProxy() {
+        delete m_colorMapper;
+    };
+
+  public slots:
```

Comment:
Instead of slots we should use the Q_INVOKABLE macro:

https://doc.qt.io/qt-5/qobject.html#Q_INVOKABLE

Slots are invoked from the event loop and therefore don't return values. The is just a semantic difference, the actual, generated implementation will probably be identical.